luaui/Widgets/cmd_extractor_snap.lua

```diff
@@ -0,0 +1,209 @@
+function widget:GetInfo()
+	return {
+		name      = "Extractor Snap (mex/geo)",
+		desc      = "Snaps extractors to give nearest spot",
+		author    = "Hobo Joe, based on work by Niobium and Floris",
+		version   = "v1.0",
+		date      = "Jan 2024",
+		license   = "GNU GPL, v2 or later",
+		layer     = 1,
+		enabled   = true,
+		handler   = true
+	}
+end
+
+
+local spGetActiveCommand = Spring.GetActiveCommand
+local spGetMouseState = Spring.GetMouseState
+local spTraceScreenRay = Spring.TraceScreenRay
+local spPos2BuildPos = Spring.Pos2BuildPos
+
+local mexConstructors
+local geoConstructors
+local mexBuildings
+local geoBuildings
+
+local selectedMex
+local selectedGeo
+local targetPos
+local cursorPos
+local buildCmd
+local unitShape
+local activeUnitShape
+local metalMap = false
+
+local isPregame = Spring.GetGameFrame() == 0 and not Spring.GetSpectatingState()
+
+local function MakeLine(x1, y1, z1, x2, y2, z2)
+	gl.Vertex(x1, y1, z1)
+	gl.Vertex(x2, y2, z2)
+end
+
+
+function widget:Initialize()
+	if not WG.DrawUnitShapeGL4 then
+		widget:Shutdown()
```

Comment:
this doesn't actually shut down the widget, just calls the function (and doesn't even stop running the `Initialize` function)
```suggestion
		widgetHandler:RemoveWidget()
		return
```

Reply:
Ah, wasn't aware of this. Thanks for the heads up.